packages/apps/job-launcher/server/src/modules/manifest/manifest.service.spec.ts

```diff
@@ -0,0 +1,795 @@
+
+        const dto = createJobCvatDto({ type: requestType });
+        const tokenFundAmount = faker.number.int({ min: 1, max: 1000 });
+
+        await expect(
+          manifestService.createManifest(dto, requestType, tokenFundAmount),
+        ).rejects.toThrow(
+          new ControlledError(ErrorJob.DataNotExist, HttpStatus.CONFLICT),
+        );
+      });
+
+      it('should throw an error if data does not exist for image skeletons from boxes job type', async () => {
+        const requestType = JobRequestType.IMAGE_SKELETONS_FROM_BOXES;
+
+        const dto = createJobCvatDto({ type: requestType });
+        const tokenFundAmount = faker.number.int({ min: 1, max: 1000 });
+
+        await expect(
+          manifestService.createManifest(dto, requestType, tokenFundAmount),
+        ).rejects.toThrow(
+          new ControlledError(ErrorJob.DataNotExist, HttpStatus.CONFLICT),
+        );
+      });
+    });
+
+    describe('createAudinoManifest', () => {
+      it('should create an Audino manifest successfully', async () => {
+        const mockDto = createJobAudinoDto(); // Use the helper function
+        const mockRequestType = JobRequestType.AUDIO_TRANSCRIPTION;
+        const mockTokenFundAmount = faker.number.int({ min: 1, max: 1000 });
+
+        const result = await manifestService.createManifest(
+          mockDto as any,
+          mockRequestType,
+          mockTokenFundAmount,
+        );
+
+        const totalSegments = Math.ceil(
+          (mockDto.audioDuration * 1000) / mockDto.segmentDuration,
+        );
+        const jobBounty =
+          ethers.parseUnits(mockTokenFundAmount.toString(), 'ether') /
+          BigInt(totalSegments);
+
+        expect(result).toEqual({
+          annotation: {
+            description: mockDto.requesterDescription,
+            labels: mockDto.labels,
+            qualifications: mockDto.qualifications || [],
+            type: mockRequestType,
+            user_guide: mockDto.userGuide,
+            segment_duration: mockDto.segmentDuration,
+          },
+          data: {
+            data_url: expect.any(String),
+          },
+          job_bounty: ethers.formatEther(jobBounty),
+          validation: {
+            gt_url: expect.any(String),
+            min_quality: mockDto.minQuality,
+          },
+        });
+      });
+    });
+
+    describe('createHCaptchaManifest', () => {
+      const requestType = JobRequestType.HCAPTCHA;
+      const tokenFundAmount = faker.number.int({ min: 1, max: 1000 });
+
+      beforeEach(() => {
+        const fileContent = JSON.stringify({
+          [faker.internet.url()]: [true, true, true],
+        });
+        (listObjectsInBucket as jest.Mock).mockResolvedValueOnce([
+          `${faker.word.sample()}.jpg`,
+          `${faker.word.sample()}.jpg`,
+          `${faker.word.sample()}.jpg`,
+        ]);
+        jest.spyOn(storageService, 'uploadJsonLikeData').mockResolvedValueOnce({
```

Comment:
You don't need `spyOn`, `storageService` is already mocked, so you can just
`storageService.uploadJsonLikeData.mockResolvedValueOnce`

If this approach is due to TS - then declare mock for storage service as variable instead of getting it as `moduleRef.get`